seqr/views/utils/orm_to_json_utils.py

```diff
@@ -757,10 +757,30 @@ def get_json_for_matchmaker_submission(submission):
         additional_model_fields=['contact_name', 'contact_href', 'submission_id'])
 
 
-def get_json_for_rna_seq_outliers(models, **kwargs):
-    additional_values = {
-        'isSignificant': Case(
-            When(p_adjust__lt=models.model.SIGNIFICANCE_THRESHOLD, then=Value(True)), default=Value(False)
-        ),
-    }
-    return get_json_for_queryset(models, additional_values=additional_values, **kwargs)
+def get_json_for_rna_seq_outliers(filters, significant_only=True):
+    filters['sample__is_active'] = True
+    data_by_individual_gene = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+
+    for model, outlier_type in [(RnaSeqOutlier, 'outliers'), (RnaSeqSpliceOutlier, 'spliceOutliers')]:
+        significant_field = model.SIGNIFICANCE_FIELD
+
+        significant_filter = {f'{significant_field}__lt': model.SIGNIFICANCE_THRESHOLD}
+        if hasattr(model, 'MAX_SIGNIFICANT_OUTLIER_NUM'):
+            significant_filter['rank__lt'] = model.MAX_SIGNIFICANT_OUTLIER_NUM
+
+        additional_values = {'isSignificant': Value(True)} if significant_only else\
+            {'isSignificant': Case(When(then=Value(True), **significant_filter), default=Value(False))}
+
+        outliers = get_json_for_queryset(
+            model.objects.filter(**filters, **(significant_filter if significant_only else {})),
+            nested_fields=[
+                {'fields': ('sample', 'tissue_type'), 'key': 'tissueType'},
+                {'fields': ('sample', 'individual', 'guid'), 'key': 'individualGuid'},
```

Comment:
if you pass `individual_guid` as a keyword arg to the function and add it as the "value" to this field, it will prevent needing to query for this field in the case where theres only one individual

Reply:
When the variant API uses this function, there is no `individual_guid`. Are different handlings for these two APIs valuable?

Reply:
This would not create different handlings, we would be using the same function for both places which would be outputting the exact same data format in both places. All we are changing is adding an optional keyword parameter that will significantly reduce the computation in the database in one case. This is consisttnt with the behavior of many functions in this file, such as:

`_get_json_for_families`: https://github.com/broadinstitute/seqr/blob/master/seqr/views/utils/orm_to_json_utils.py#L242
`get_json_for_samples`: https://github.com/broadinstitute/seqr/blob/master/seqr/views/utils/orm_to_json_utils.py#L340-L344
`get_json_for_matchmaker_submissions`: https://github.com/broadinstitute/seqr/blob/master/seqr/views/utils/orm_to_json_utils.py#L742

Reply:
Okay. I will remove the `individual_guid` nested field for the individual API.